### PR TITLE
Add EN Blog: Restart a Kubernetes Cluster

### PR DESCRIPTION
**Pull request by Felixnoo**
Signed-off-by: Felixnoo <felixliu@kubesphere.io>

#2335 

/cc @kubesphere/sig-installation 
This blog tries to provide a generic idea for restarting a Kubernetes cluster. Please give a technical review on this blog, especially on the `Restart a Kubernetes Cluster` part (line 21-138). -> [Preview link](https://deploy-preview-2345--kubesphere-v3-1.netlify.app/blogs/restart-k8s-cluster/).

### CODE REVIEW

**Comment by Felixnoo** [comment]
/cc @kubesphere/sig-installation 
Hi, could you help review this PR? Thanks a lot.

**Comment by Felixnoo** [comment]
/cc @kubesphere/sig-installation 
Hi, could you help review this PR? The reviewer will be treated as a co-author of this blog article. Thanks a lot.

File: content/en/blogs/restart-k8s-cluster.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+   ```
+   kubectl cordon <worker node name>
+   kubectl drain <worker node name>
+   ```
+
+3. Run the following command to stop kubelet and kube-proxy.
+
+   ```
+   sudo docker stop kubelet kube-proxy
```
Comment (pixiake) [inline content/en/blogs/restart-k8s-cluster.md:39]: Stop kubelet can use `systemctl stop kubelet`.
`kube-proxy` run in a cluster as pods and no processing is required.

File: content/en/blogs/restart-k8s-cluster.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+1. Connect to a control plane through SSH.
+
+2. Run the following commands to stop pod scheduling and drain existing pods on the node.
+
+   ```
+   kubectl cordon <control plane name>
+   kubectl drain <control plane name>
+   ```
+
+3. Run the following command to stop kubelet and kube-proxy.
```
Comment (pixiake) [inline content/en/blogs/restart-k8s-cluster.md:67]: Same as above

File: content/en/blogs/restart-k8s-cluster.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+   ```
+   sudo docker stop kubelet kube-proxy
+   ```
+
+4. Run the following command to stop kube-scheduler and kube-controller-manager.
+
+   ```
+   sudo docker stop kube-scheduler kube-controller-manager
```
Comment (pixiake) [inline content/en/blogs/restart-k8s-cluster.md:76]: `kube-apiserver` `kube-scheduler` `kube-controller-manager` run in a cluster as pods and no processing is required.

File: content/en/blogs/restart-k8s-cluster.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+   sudo docker stop kube-apiserver
+   ```
+
+6. Run the following command to stop Docker.
+
+   ```
+   sudo systemctl stop docker
+   ```
+
+7. Run the following command to shut down the control plane.
```
Comment (pixiake) [inline content/en/blogs/restart-k8s-cluster.md:91]: If etcd is deployed on the master node, the etcd service needs to be stopped first. `systemctl stop etcd`
If etcd run in a cluster as pod and no processing is required.

File: content/en/blogs/restart-k8s-cluster.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+8. Perform the same operations on other control planes (if any) to shut them down.
+
+### Shut down ectd nodes
+
+1. Connect to an etcd node through SSH.
+
+2. Run the following command to stop kubelet and kube-proxy.
+
+   ```
+   sudo docker stop kubelet kube-proxy
```
Comment (pixiake) [inline content/en/blogs/restart-k8s-cluster.md:106]: Same as above

File: content/en/blogs/restart-k8s-cluster.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+   ```
+
+3. Run the following command to stop etcd.
+
+   ```
+   sudo docker stop etcd
```
Comment (pixiake) [inline content/en/blogs/restart-k8s-cluster.md:112]: `systemctl stop etcd`

**Comment by pixiake** [comment]
I have added some comments. However, different cluster installation methods may have different ways to restart the cluster. My comments are only for clusters installed with kubekey.

File: content/en/blogs/restart-k8s-cluster.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+### Shut down worker nodes
+
+1. Connect to a worker node through SSH.
+
+2. Run the following commands to stop pod scheduling and drain existing pods on the node.
+
+   ```
+   kubectl cordon <worker node name>
+   kubectl drain <worker node name>
```
Comment (pixiake) [inline content/en/blogs/restart-k8s-cluster.md:33]: This command may need to add parameters (`--delete-emptydir-data` `--ignore-daemonsets`) to evict all pods correctly.
```
kubectl drain <worker node name> --ignore-daemonsets --delete-emptydir-data
```

**Comment by Felixnoo** [comment]
> I have added some comments. However, different cluster installation methods may have different ways to restart the cluster. My comments are only for clusters installed with kubekey.

@pixiake 
Hi, I made [updates](https://github.com/kubesphere/website/pull/2345/commits/d9313360ac12a9c0a5bcfaff8e38a02de433e42f) according to your comments. Please take a look again when you are available.

**Comment by Felixnoo** [comment]
/assign @JiannaLiu @FeynmanZhou 
Please take a look at this PR when you are available, thanks.

**Pull request merged**